addons/web/static/tests/fields/basic_fields_tests.js

```diff
@@ -479,6 +479,87 @@ QUnit.module('basic_fields', {
+
+        this.data.partner.fields.float_time_field = { string: "Float Time", type: 'float_time' };
+        this.data.partner.records[0].float_time_field = 9.99;
+
+        this.data.partner.fields.monetary = { string: "Monetary", type: 'monetary' };
+        this.data.partner.records[0].monetary = 9.99;
+        this.data.partner.records[0].currency_id = 1;
+
+        this.data.partner.fields.percentage = { string: "Percentage", type: 'percentage' };
+        this.data.partner.records[0].percentage = .99;
+
+        const form = await createView({
+            View: FormView,
+            model: 'partner',
+            data: this.data,
+            arch: `
+                <form string="Partners">
+                    <field name="float_factor_field" options="{'factor': 0.5}"/>
+                    <field name="qux"/>
+                    <field name="float_time_field"/>
+                    <field name="int_field"/>
+                    <field name="monetary"/>
+                    <field name="currency_id" invisible="1"/>
+                    <field name="percentage"/>
+                </form>
+            `,
+            res_id: 1,
+            translateParameters: {
+                decimal_point: "🇧🇪",
+            },
+        });
+
+        // Prepare the numpad "dot" keydown event
+        const ev = new KeyboardEvent('keydown', { code: 'NumpadDecimal' });
+        ev.getModifierState = (mod) => mod === 'NumLock';
+
+        // Record edit mode
+        await testUtilsDom.click(form.el.querySelector('.o_form_button_edit'));
+
+        // Dispatch to all inputs and check
+        const floatFactorField = form.el.querySelector('.o_input[name="float_factor_field"]');
+        const floatInput = form.el.querySelector('.o_input[name="qux"]');
+        const floatTimeInput = form.el.querySelector('.o_input[name="float_time_field"]');
+        const integerInput = form.el.querySelector('.o_input[name="int_field"]');
+        const monetaryInput = form.el.querySelector('.o_input[name="monetary"]');
+        const percentageInput = form.el.querySelector('.o_input[name="percentage"]');
+
+        floatFactorField.dispatchEvent(ev);
+        await testUtils.nextTick();
+        assert.ok(floatFactorField.value.endsWith('🇧🇪'));
+
+        floatInput.dispatchEvent(ev);
+        await testUtils.nextTick();
+        assert.ok(floatInput.value.endsWith('🇧🇪'));
+
+        floatTimeInput.dispatchEvent(ev);
+        await testUtils.nextTick();
+        assert.ok(floatTimeInput.value.endsWith('🇧🇪'));
+
+        integerInput.dispatchEvent(ev);
+        await testUtils.nextTick();
+        assert.ok(integerInput.value.endsWith('🇧🇪'));
+
+        monetaryInput.dispatchEvent(ev);
+        await testUtils.nextTick();
+        assert.ok(monetaryInput.childNodes[0].value.endsWith('🇧🇪'));
+
+        percentageInput.dispatchEvent(ev);
+        await testUtils.nextTick();
+        assert.ok(percentageInput.childNodes[0].value.endsWith('🇧🇪'));
```

Comment:
prefer monetaryInput.querySelector([a selector to the float input])
and percetageInput.querySelector.......
For clarity when reading the test, and be more robust if we were to change the field's dom